be/src/env/env_s3.cpp

```diff
@@ -77,15 +77,20 @@ StatusOr<std::unique_ptr<RandomAccessFile>> EnvS3::new_random_access_file(const
     std::string object = path.substr(namenode.size(), path.size() - namenode.size());
     Aws::Client::ClientConfiguration config;
     config.scheme = Aws::Http::Scheme::HTTP;
-    if (!config::aws_s3_endpoint.empty()) {
-        config.endpointOverride = config::aws_s3_endpoint;
+    if (is_oss_path(namenode.c_str())) {
+        assert(!config::object_storage_endpoint.empty());
+        config.endpointOverride = get_endpoint_from_oss_bucket(config::object_storage_endpoint, &bucket);
+    }
+    if (is_s3a_path(namenode.c_str()) && !config::object_storage_endpoint.empty()){
+        config.endpointOverride = config::object_storage_endpoint;
+    }
+    config.maxConnections = config::object_storage_max_connection;
+    if (is_oss_path(namenode.c_str())) {
+        assert(!config::object_storage_access_key_id.empty());
```

Comment:
use `CHECK` to replace `assert`. And combine to comments before, you could write a separate logic for oss, I think

```
if (oss_path() {
.....
} else {
....
}
```

Reply:
done, merged with endpoint check